components/inputs/test/input-number.test.js

```diff
@@ -387,6 +387,14 @@ describe('d2l-input-number', () => {
 			await aTimeout(1);
 
 			expect(fired).to.be.false;
+
+		});
+
+		it('should have correct "invalid" state when change event fires', async() => {
+			const elem = await fixture(minMaxFixture);
+			setTimeout(() => setInnerInputValue(elem, '0'));
+			await oneEvent(elem, 'change');
```

Comment:
I'm just noticing that `setInnerInputValue` itself fires a change event on the element:

```js
setTimeout(() => dispatchEvent(inputTextElement, 'change'));
```

Is this line waiting for that event, or for the one fired in input-number's `_handleChange()`? Seems like there could be a race condition there.

Reply:
It's a different event. We're simulating the user modifying the underlying `<d2l-input-text>`'s value (which in turn sets an underlying `<input type="text">`) and firing a change on that `<d2l-input-text>`. That's what [the number input is listening for](https://github.com/BrightspaceUI/core/blob/main/components/inputs/input-number.js#L326) and will trigger the logic in `_handleChange` to run, which in turn fires a `change` event on the actual `d2l-input-number`, which is what the test is waiting for here.